grails-app/services/org/pih/warehouse/inventory/InventoryService.groovy

```diff
@@ -2733,11 +2734,19 @@ class InventoryService implements ApplicationContextAware {
             return []
         }
 
-        def criteria = TransactionEntry.createCriteria()
-        return criteria.list {
+        return TransactionEntry.createCriteria().list {
             'in'("inventoryItem", inventoryItems)
             transaction {
-                eq("transactionDate", date)
+                // Catch transactions that are created within the same minute
+                // We are saving the adjustment transaction one second after
+                // inventory baseline, so to get that transaction, we should
+                // match transaction based on the time range.
+                // For a date: 18.07.2025 15:28:00 we are matching transactions
+                // that are created: 18.07.2025 15:28:00 <= date < 18.07.2025 15:28:59
+                // We are able to select dates with the precision set to minutes,
+                // so we should have at most one transaction within the same minute
+                ge("transactionDate", date)
```

Comment:
You can use `between` I think to make this a single line.

https://docs.grails.org/latest/ref/Domain%20Classes/createCriteria.html

But I still don't think this is necessary. See my other comment